javascript/extractor/lib/typescript/src/main.ts

```diff
@@ -104,7 +107,7 @@ function checkCycle(root: any) {
         obj.$cycle_visiting = true;
         for (let k in obj) {
             if (!obj.hasOwnProperty(k)) continue;
-            if (isBlacklistedProperty(k)) continue;
+            if (+k !== +k && !astPropertySet.has(k)) continue;
```

Comment:
the `+k !== +k` is whitelisting all the number properties. I think that deserves a small comment.